plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -19,50 +19,68 @@
 
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.automaton.CharacterRunAutomaton;
 import org.apache.tika.language.LanguageIdentifier;
 import org.apache.tika.metadata.Metadata;
 import org.apache.tika.metadata.TikaCoreProperties;
 import org.elasticsearch.ElasticsearchParseException;
 import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.common.logging.ESLoggerFactory;
 import org.elasticsearch.ingest.AbstractProcessor;
 import org.elasticsearch.ingest.IngestDocument;
 import org.elasticsearch.ingest.Processor;
 
 import java.io.IOException;
-import java.util.Arrays;
 import java.util.EnumSet;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
 
-import static org.elasticsearch.ingest.ConfigurationUtils.newConfigurationException;
+import static org.elasticsearch.common.regex.Regex.simpleMatchToAutomaton;
 import static org.elasticsearch.ingest.ConfigurationUtils.readBooleanProperty;
 import static org.elasticsearch.ingest.ConfigurationUtils.readIntProperty;
 import static org.elasticsearch.ingest.ConfigurationUtils.readOptionalList;
 import static org.elasticsearch.ingest.ConfigurationUtils.readStringProperty;
+import static org.elasticsearch.ingest.attachment.AttachmentProcessor.ReservedProperty.findDeprecatedProperty;
 
 public final class AttachmentProcessor extends AbstractProcessor {
 
+    private static final Logger logger = ESLoggerFactory.getLogger(AttachmentProcessor.class);
+    private static final DeprecationLogger deprecationLogger = new DeprecationLogger(logger);
+
     public static final String TYPE = "attachment";
 
     private static final int NUMBER_OF_CHARS_INDEXED = 100000;
 
     private final String field;
     private final String targetField;
-    private final Set<Property> properties;
+    private final Set<ReservedProperty> reservedProperties;
+    private final Set<String> properties;
     private final int indexedChars;
     private final boolean ignoreMissing;
+    private final CharacterRunAutomaton runAutomaton;
 
-    AttachmentProcessor(String tag, String field, String targetField, Set<Property> properties,
-                        int indexedChars, boolean ignoreMissing) throws IOException {
+    AttachmentProcessor(String tag, String field, String targetField, Set<ReservedProperty> reservedProperties,
+                        Set<String> properties, int indexedChars, boolean ignoreMissing) throws IOException {
         super(tag);
         this.field = field;
         this.targetField = targetField;
         this.properties = properties;
+        this.reservedProperties = reservedProperties;
         this.indexedChars = indexedChars;
         this.ignoreMissing = ignoreMissing;
+
+        // We build the regex automaton we will use to extract raw metadata
+        if (properties.isEmpty()) {
```

Comment:
if you put the standard fields into the automaton as well, you could construct the automaton in the factory and just hand it over as a parameter?